packages/svelte-cloudinary/src/components/CldOgImage.svelte

```diff
@@ -1,74 +1,87 @@
-	import type { CldImageProps } from './CldImageTypes.ts';
-	import { OG_IMAGE_WIDTH, OG_IMAGE_WIDTH_RESIZE, OG_IMAGE_HEIGHT } from '../constants/sizes.js';
+<!--
+	@component
+	
+	This component will use Cloudinary to power your open graph images (and twitter).
+	It'll automatically generate your meta tags and place them in the head of your
+	page.
 
-	const TWITTER_CARD = 'summary_large_image';
+	@see https://svelte.cloudinary.dev/components/og-image
 
-	type CldOgImageProps = Omit<CldImageProps, 'heigth' | 'heigth' | 'layout'> & {
-		excludeTags?: Array<string>;
-		keys?: object;
-		twitterTitle?: string;
-		width?: string | number;
-		height?: string | number;
-	};
-	type $$Props = CldOgImageProps;
+	@example Simple OG Image
 
-	const options: ImageOptions = {
-		...$$props,
-		crop: $$props.crop || 'fill',
-		gravity: $$props.gravity || 'center',
-		height: $$props.height || OG_IMAGE_HEIGHT,
-		src: $$props.src,
-		width: $$props.width || OG_IMAGE_WIDTH,
-		widthResize: $$props.width || OG_IMAGE_WIDTH_RESIZE
-	};
+	```svelte
+	<script>
+		import { CldOgImage } from 'svelte-cloudinary';
+	</script>
 
-	let width = typeof options.width === 'string' ? parseInt(options.width) : options.width;
-	let height = typeof options.height === 'string' ? parseInt(options.height) : options.height;
-	let { alt, excludeTags = [], twitterTitle } = $$props;
+	<CldOgImage
+		src="images/turtle"
+		alt="Turtle"
+	/>
+	```
+-->
 
-	// Resize the height based on the widthResize property
+<script context="module" lang="ts">
+	import type {
+		ConfigOptions,
+		ImageOptions,
+	} from '@cloudinary-util/url-loader';
 
-	if (typeof height === 'number' && typeof width === 'number') {
-		height = (OG_IMAGE_WIDTH_RESIZE / width) * height;
-	}
+	export type CldOgImageProps = ImageOptions & {
+		/**
+		 * The title to use on twitter
+		 */
+		twitterTitle?: string;
 
-	width = OG_IMAGE_WIDTH_RESIZE;
+		/**
+		 * The image alt text
+		 */
+		alt: string;
 
-	// Render the final URLs. We use two different format versions to deliver
-	// webp for Twitter as it supports it (and we can control with tags) where
-	// other platforms may not support webp, so we deliver jpg
+		/**
+		 * The width of your og image
+		 * @default 1200
+		 */
+		width?: number;
 
-	const ogImageUrl = getCldImageUrl({
-		...options,
-		format: $$props.format || 'jpg'
-	});
+		/**
+		 * The height of your og image
+		 * @default 627
+		 */
+		height?: number;
 
-	const twitterImageUrl = getCldImageUrl({
-		...options,
-		format: $$props.format || 'webp'
-	});
+		/**
+		 * Overrides for the global Cloudinary config.
+		 * @see https://svelte.cloudinary.dev/config
+		 */
+		config?: ConfigOptions;
+	};
 </script>
 
-<svelte:head>
-	<meta property="og:image" content={ogImageUrl} />
-	<meta property="og:image:secure_url" content={ogImageUrl} />
-	<meta property="og:image:width" content={`${width}`} />
-	<meta property="og:image:height" content={`${height}`} />
+<script lang="ts">
```

Comment:
looks like this is currently missing the different versions of the OG images for the base og:image and the twitter:image

twitter allows you to use webp images, which is a more optimal image format, where most other places don't, but because we can specify a separate twitter image, we can custom configure the twitter version as webp. can see it in the diff

Reply:
https://github.com/cloudinary-community/svelte-cloudinary/issues/127